Modules/Sources/Yosemite/Model/Storage/PersistedProduct+Conversions.swift

```diff
@@ -42,11 +42,47 @@ extension PersistedProduct {
             stockStatusKey: stockStatusKey
         )
     }
+
+    func toPOSProduct(db: GRDBDatabaseConnection) throws -> POSProduct {
+        let images = try db.read { db in
+            return try request(for: PersistedProduct.images).fetchAll(db)
+        }
+        let attributes = try db.read { db in
+            return try request(for: PersistedProduct.attributes).fetchAll(db)
+        }
+        return toPOSProduct(
+            images: images.map { $0.toProductImage() },
+            attributes: attributes.map { $0.toProductAttribute(siteID: siteID) }
+        )
+    }
+
+}
+
+// MARK: - POSProduct Storage Extensions
+extension POSProduct {
+    public func save(to db: GRDBDatabaseConnection) throws {
+        try db.write { db in
```

Comment:
do you know if it's recommended to weak self in the `db.write` closure?

Reply:
I don't know of a recommendation.

The `updates` closure that we pass to `write` is non-escaping, so it can't be stored by GRDB, and we don't hold it beyond our function scope either. 

I don't see any reason to make `self` weak here, at least not to avoid retain cycles.